src/cca.jl

```diff
@@ -336,3 +340,113 @@ function fit(::Type{CCA}, X::AbstractMatrix{T}, Y::AbstractMatrix{T};
 
     return M::CCA
 end
+
+abstract type MultivariateTest <: HypothesisTest end
+
+struct WilksLambda <: MultivariateTest
+    stat::Float64
+    fstat::Float64
+    df1::Float64
+    df2::Float64
+end
+
+struct LawleyHotelling <: MultivariateTest
+    stat::Float64
+    fstat::Float64
+    df1::Float64
+    df2::Float64
+end
+
+struct PillaiTrace <: MultivariateTest
+    stat::Float64
+    fstat::Float64
+    df1::Float64
+    df2::Float64
+end
+
+function pvalue(ct::MultivariateTest)
+    return 1 - cdf(FDist(ct.df1, ct.df2), ct.fstat)
+end
+
+function dof(ct::MultivariateTest)
+    return (ct.df1, ct.df2)
+end
+
+function _testprep(cca::CCA, n, k)
+
+    r = cca.eigs[k:end]
+    dx = length(cca.xmean)
+    dy = length(cca.ymean)
+    if isnothing(n) && cca.nobs == -1
+        @error("If CCA was fit using :cov, n must be provided to tests")
```

Comment:
It's generally unusual for functions like this to log rather than throw errors. Is there a reason why this shouldn't be a proper error?

Reply:
Done